src/Apps/Artist/Routes/Overview/Components/ArtworkFilter/index.tsx

```diff
@@ -408,14 +409,38 @@ export const ArtworkFilterFragmentContainer = createFragmentContainer(
           artworks
           has_make_offer_artworks
         }
-        filtered_artworks(aggregations: $aggregations, size: 0) {
+        # The call to fetch aggregations shouldn't take any
+        # filter params (since we always want the full list for an artist).
+        filtered_artworks(
+          sort: $sort
+          page: $page
+          aggregations: $aggregations
+        ) {
           aggregations {
             slice
             counts {
               name
               id
             }
           }
+          # Include the below fragment so that this will match
+          # the initial load (w/ no filter applied), and thus MP
+          # will consolidate aggregations _and_ the grid into one call.
+          # Leave out this fragment if navigating to the artist page
+          # with a filter applied, as those can't be consolidated and
+          # this is extra data.
+          #
+          # TODO: Metaphysics will still make this backend request, just not
+          # return any data if it's skipped. Improve introspection, as this
+          # will overfetch in cases of navigating to the artist page with
+          # a filter applied.
+          artworks_connection(first: 24, after: "") @skip(if: $hasFilter) {
```

Comment:
The trick here, is that this winds up being the same request as the grid, and so everything gets consolidated by Metaphysics.

However, if a filter is applied, while this fragment isn't returned to the client due to the `skip` directive, it appears as though Metaphysics _is_ still making a backend request for these artworks. In the case of hitting the page w/ a filter applied, this will actually _overfetch_ on the backend.

So to correct that (which is what the TODO here is referring to), I'd like to improve the resolver to be able to better introspect and determine if the connection is 'skipped'. If so, it can make sure `size=0` goes out w/ any backend requests.